src/officehours_api/tests.py

```diff
@@ -163,3 +169,92 @@ def test_meeting_start_notifies_attendees(self, mock_twilio: mock.MagicMock):
+        data = {
+            'queue': self.zoom_queue.id,
+            'attendee_ids': [self.user2.id],
+            'agenda': 'test agenda',
+            'assignee_id': self.user1.id,
+            'backend_type': 'inperson'
+        }
+        serializer = MeetingSerializer(data=data)
+        with self.assertRaises(ValidationError) as cm:
+            serializer.is_valid(raise_exception=True)
+        error = str(cm.exception.detail['non_field_errors'][0])
+        self.assertEqual(error, "inperson is not one of the queue's allowed backend types (['zoom'])")
+
+    def test_backend_type_valid(self):
+        data = {
+            'queue': self.inperson_queue.id,
+            'attendee_ids': [self.user2.id],
+            'agenda': 'test agenda',
+            'assignee_id': self.user1.id,
+            'backend_type': 'inperson'
+        }
+        serializer = MeetingSerializer(data=data)
+        valid = serializer.is_valid(raise_exception=False)
+        self.assertTrue(valid)
+
+    def test_patch_backend_type_none(self):
+        data = {
+            'queue': self.inperson_queue.id,
+            'attendee_ids': [self.user2.id],
+            'agenda': 'test agenda',
+            'assignee_id': self.user1.id,
+            'backend_type': 'inperson'
+        }
+        serializer = MeetingSerializer(data=data)
+        valid = serializer.is_valid(raise_exception=False)
+        self.assertTrue(valid)
+        # Patch the meeting without a backend type
+        data = {
+            'queue': self.inperson_queue.id,
+            'attendee_ids': [self.user2.id],
+            'agenda': 'test agenda',
+            'assignee_id': self.user1.id,
+        }
+        serializer = MeetingSerializer(data=data)
+        valid = serializer.is_valid(raise_exception=False)
+        self.assertTrue(valid)
+
+    @skipIf('zoom' not in ENABLED_BACKENDS, 'Skipping because "zoom" backend type is not enabled')
+    def test_patch_backend_type_invalid(self):
+        data = {
+            'queue': self.inperson_queue.id,
+            'attendee_ids': [self.user2.id],
+            'agenda': 'test agenda',
+            'assignee_id': self.user1.id,
+            'backend_type': 'inperson'
+        }
+        serializer = MeetingSerializer(data=data)
+        valid = serializer.is_valid(raise_exception=False)
+        self.assertTrue(valid)
+        # Patch the meeting to an invalid backend type
+        data = {
+            'queue': self.inperson_queue.id,
+            'attendee_ids': [self.user2.id],
+            'agenda': 'test agenda',
+            'assignee_id': self.user1.id,
+            'backend_type': 'zoom'
+        }
+        serializer = MeetingSerializer(data=data)
+        with self.assertRaises(ValidationError) as cm:
+            serializer.is_valid(raise_exception=True)
+        error = str(cm.exception.detail['non_field_errors'][0])
+        self.assertEqual(error, "zoom is not one of the queue's allowed backend types (['inperson'])")
```

Comment:
I don't know if this is really testing what happens a meeting is patched. I think we need to something more like what is described here: https://www.django-rest-framework.org/api-guide/serializers/#partial-updates

I think it could be something like this, where you create a model instance and then only pass the updated data.
```
        meeting = Meeting(**{
            'queue': self.inperson_queue.id,
            'attendee_ids': [self.user2.id],
            'agenda': 'test agenda',
            'assignee_id': self.user1.id,
            'backend_type': 'inperson'
        })
        # Patch the meeting to an invalid backend type
        data = {
            'backend_type': 'zoom'
        }
        serializer = MeetingSerializer(meeting, data=data, partial=True)
```
Can we use this approach for the two patch cases?

Reply:
We tried to achieve something like this but failed to find this partial-updates documents. Thank you so much for the updates! We have added it to our latest commit `d0e34cc`!
For the patch test cases, now we would only assert once after the meeting is patched. So the flow would be to create a new meeting, patch the meeting with new info, and run the validation.